contract/contract.go

```diff
@@ -100,7 +100,8 @@ func Execute(execCtx context.Context, bs *state.BlockState, cdb ChainAccessor, t
 
 	// compute gas limit
 	var gasLimit uint64
-	if gasLimit, err = GasLimit(bi.ForkVersion, isFeeDelegation, txGasLimit, len(txPayload), bs.GasPrice, usedFee, sender.Balance(), receiver.Balance()); err != nil {
+	if gasLimit, err = fee.GasLimit(bi.ForkVersion, isFeeDelegation, txGasLimit, len(txPayload), bs.GasPrice, usedFee, sender.Balance(), receiver.Balance()); err != nil {
+		err = newVmError(types.ErrNotEnoughGas)
```

Comment:
Why is it overwriting the `err` here?

Reply:
this is confusing code. I will fix it.

Reply:
No big problem, I understood now that the called fn is not using "types" and NewVmError